dask/dataframe/io.py

```diff
@@ -441,8 +441,12 @@ def to_hdf(df, path_or_buf, key, mode='a', append=False, complevel=0,
 
     dask_kwargs = dask_kwargs or {}
 
-    DataFrame._get(merge(df.dask, dsk), (name, df.npartitions - 1),
-                   get=get, **dask_kwargs)
+    dsk = merge(df.dask, dsk)
+    keys = (name, df.npartitions - 1)
+    if compute:
+        DataFrame._get(dsk, keys, get=get, **dask_kwargs)
+    else:
+        return dsk, keys
```

Comment:
Bag.to_textfiles also returns a db.Bag (https://github.com/dask/dask/blob/master/dask/bag/core.py#L184).

dd.dataframe.tail and head also return specialized objects and not dask graphs.

Looks like we should conform to returning dask,Delayed.

Other opinions?